FoodOrderingSystem.py

```diff
@@ -0,0 +1,843 @@
+        for fc, fd in menu:
+            print("\nFood ID: {} \nFood category: {} \nFood name: {} \nFood price: {}".format(fd.food_id, fc.name, fd.food_name, fd.price))
+
+    def customer_signup(self, cust_name, cust_phone, cust_email, session):
+        c = self.customer.customer_signup(cust_name, cust_phone, cust_email, session)
+        if c:
+            print("\nCustomer ID: {} \nCustomer name: {} \nCustomer phone number: {} \nCustomer email address: {} \nSucessfully!".format(c.cust_id, c.cust_name, c.cust_phone, c.cust_email))
+        else:
+            print("Adding details was not sucessfully. Please try again!")    
+
+
+class DeliveryPerson(Base, Employee):
+    """ Represents delivery person """
+
+    __tablename__ = 'delivery_person'
+
+    delivery_person_id = Column(Integer(), primary_key=True, nullable=False, unique=True, autoincrement=True)
+    delivery_person_name = Column(String(30), nullable=False)
+    delivery_person_phone = Column(Integer(), nullable=False)
+
+    def update_order(self, order_id, order_status, session):
+        """ Delivery person can update the order """
+
+        super(DeliveryPerson, self).__init__()
+
+        update = session.query(
+            CustOrderStatus
+            ).filter(CustOrderStatus.order_id == order_id
+            ).update(
+                {CustOrderStatus.order_status: order_status}, 
+                synchronize_session=False
+                )
+        try:
+            session.commit()
+            return update
+        except:
+            session.rollback()
+            raise Exception("update not completed!")
+        finally:
+            session.expunge_all()
+            session.close()
+
+def main():
+    """ The user interface in which employee/customer will use to perform actions on the Food Ordering System. """
+
+    db_url = "sqlite:///fos.db"
+    print("Connecting to {}".format(db_url))
+
+    fos = Controller(db_url)
+    fos.bootstrap()
+
+    welcome_message = """ 
+    Welcome to the Food Ordering System! Please press the below options:
+
+    1. Employee
+    2. Customer
+
+    Your Option:
+
+    """
+
+    option = int(input(welcome_message))
+
+    if option == 1:
+        emp_options = """         
+        Welcome to the employee interface! Please press the below options:
+        
+        0. Quit
+        1. Add food category
+        2. Add food details
+        3. Add delivery person
+        4. Assign delivery person to order 
+        5. Update order (Only for delivery person use)
+        6. View order details
+        7. View order status
+        8. View revenue today
+        9. Delete order
+    
+        Your Option:
+
+        """
+
+        employee_options = int(input(emp_options))
+
+        while employee_options != 0:
+
+            if employee_options == 1:
+                category_name = input("Enter category name: ")
+                session = fos.Session()
+                fos.add_food_category(category_name, session)
```

Comment:
REFACTOR:

You can actually create the session object inside the Controller's add_food_category(..) function because the session belongs to the Controller's class already, so the main(..) function doesn't need to have the logic for the session stuff. Keep it simple!

Reply:
Do the same for all remaining functions.

Reply:
Change made to session object.